### PR TITLE
Jv-fruit-shop task solution

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

File: src/main/java/core/basesyntax/Main.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+public class Main {
+    public static void main(String[] args) {
+        IFileReader cvsReader = new FileReaderCvs();
+        IFileWriter cvsWriter = new FileWriterCvs();
+        EventHandler eventHandler = new EventHandler();
+
+        var list = cvsReader.read("src/main/java/core/basesyntax/resources/activities.csv");
+        eventHandler.convertList(list);
+        cvsWriter.write("src/main/java/core/basesyntax/resources/report.csv");
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Main.java:17]: - do not use **var**
- use constants for file names
- 
image

- see main() method structure and implement it
image

File: src/main/java/core/basesyntax/model/Event.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package core.basesyntax.model;
+
+public record Event(Operation operation, String name, int quantity) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/model/Event.java:3]: Use terminology from the task. Do not invent new words (like Event)

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/core/basesyntax/Main.java (hunk 1 above)
Comment (mkazarian-1) [reply src/main/java/core/basesyntax/Main.java]: Sorry for the first commit, I didn't see the additional explanation in the README file

File: src/main/java/core/basesyntax/dao/IFruitTransactionDao.java
Comment (kerrrusha) [inline src/main/java/core/basesyntax/dao/IFruitTransactionDao.java:1]: looks like you have experience with C#) I also used to name interfaces with "I" prefix in C#, but in Java we dont do like that. kindly asking you to remove this prefix from all interface names

File: src/main/java/core/basesyntax/service/impl/FileReaderCvs.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.IFileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class FileReaderCvs implements IFileReader {
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/service/impl/FileReaderCvs.java:9]:
```suggestion
public class FileReaderCsv implements IFileReader {
```

File: src/main/java/core/basesyntax/service/impl/FileReaderCvs.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Override
+    public List<String> read(String path) {
+        List<String> lineList;
+        try {
+            lineList = Files.readAllLines(Path.of(path));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't find file by path: " + path, e);
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/service/impl/FileReaderCvs.java:17]: its not guaranteed that exception will be related only to not finding file
```suggestion
            throw new RuntimeException("Can't read file by path: " + path, e);
```

File: src/main/java/core/basesyntax/service/impl/FileWriterCvs.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.IFileWriter;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class FileWriterCvs implements IFileWriter {
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/service/impl/FileWriterCvs.java:8]:
```suggestion
public class FileWriterCsv implements IFileWriter {
```

File: src/main/java/core/basesyntax/service/impl/FileWriterCvs.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+    @Override
+    public void write(String path, String text) {
+        try (BufferedWriter buf = new BufferedWriter(new FileWriter(path, false))) {
+            buf.write(text);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't find file by path: " + path, e);
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/service/impl/FileWriterCvs.java:15]: its not guaranteed that exception will be related only to not finding file
```suggestion
            throw new RuntimeException("Can't read file by path: " + path, e);
```

File: src/main/java/core/basesyntax/dao/IFruitTransactionDao.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.model.FruitTransaction;
+
+public interface IFruitTransactionDao {
+    void add(FruitTransaction fruitStorage);
+
+    FruitTransaction getByIndex(int index);
+
+    int getLength();
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/dao/IFruitTransactionDao.java:10]: better name
```suggestion
    int getCount();
```

File: src/main/java/core/basesyntax/db/Storage.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.db;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    public static final List<FruitTransaction> fruitTransactions = new ArrayList<>();
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/db/Storage.java:8]: using HashMap with key of String where key is fruit name is much more optimized and faster than working with FruitTransaction objects in List
```suggestion
    public static final Map<String, Integer> fruits = new HashMap<>();
```

**Review by kerrrusha — CHANGES_REQUESTED** [review]
overall very good job!

**Review by kerrrusha — APPROVED** [review]

**Pull request closed without merging**